url-parameters/tests.json

```diff
@@ -0,0 +1,39 @@
+{
+    "trackingParameters": {
+        "name": "Tracking parameters are able to be removed from a URL",
+        "desc": "Those tests use config_reference.json",
+        "referenceConfig": "config_reference.json",
+        "tests": [
+            {
+                "name": "Remove utm tracking parameters from URL",
+                "testURL": "http://www.example.com/test.html?utm_source=test&utm_medium=test&utm_campaign=test&utm_term=test&utm_content=test",
+                "expectURL": "http://www.example.com/test.html",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Remove fbclid parameter and retain other parameters",
+                "testURL": "http://www.example.com/test.html?fbclid=test&q=test&id=5",
+                "expectURL": "http://www.example.com/test.html?q=test&id=5",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Retain all non tracking parameters",
+                "testURL": "http://www.example.com/test.html?q=test&id=5",
+                "expectURL": "http://www.example.com/test.html?q=test&id=5",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Don't remove parameters because domain is in exceptions list",
+                "testURL": "http://example2.com/test.html?gclid=test&fbclid=test",
+                "expectURL": "",
```

Comment:
this doesn't seem right - should be same URL looking at the test above, right?

Reply:
I could make an exception in the platform test implementation but the current implementation of the feature returns nil if there is no change to the URL.

Reply:
Ok slight correction. If a URL is excluded (i.e. user exception, config exception) it returns nil. If a URL is not excluded but not rewritten we get the original URL.

Reply:
Gotcha, but this is very iOS specific and other implementation can return unchanged URL in this case. What we expect is the URL to be exactly the same so I think `expectURL` should reflect that. Instead of having empty string here, you should change iOS integration to sth like:

```js

    $cleanURL = LinkCleaner.removeTrackingParameters($testURL)

    if ($cleanURL === nil) { // url unchanged
       $cleanURL = $testURL;
    }

     expect($cleanURL === $test.expectURL)